### ISSUE TITLE
Fields in a NamedTuple are implicitly read-only for purposes of protocol matching

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#5683]
The following code should generate an error but doesn't.

```python
from typing import Protocol, NamedTuple


class Proto(Protocol):
    a: str


class B(NamedTuple):
    a: str


# This should generate an error because the protocol
# indicates that 'a' must be writable.
x: Proto = B("")
```

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.323, which I just published. It will also be included in a future release of pylance.
